src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -812,29 +819,39 @@ FullQName AdvertiserMinMdns::GetCommissioningTxtEntries(const CommissionAdvertis
                                                                                            : &mQueryResponderAllocatorCommissioner;
 
     char txtVidPid[chip::Dnssd::kKeyVendorProductMaxLength + 4];
-    if (params.GetProductId().HasValue() && params.GetVendorId().HasValue())
-    {
-        snprintf(txtVidPid, sizeof(txtVidPid), "VP=%d+%d", params.GetVendorId().Value(), params.GetProductId().Value());
-        txtFields[numTxtFields++] = txtVidPid;
-    }
-    else if (params.GetVendorId().HasValue())
     {
-        snprintf(txtVidPid, sizeof(txtVidPid), "VP=%d", params.GetVendorId().Value());
-        txtFields[numTxtFields++] = txtVidPid;
+        const auto & productId = params.GetProductId();
+        const auto & vendorId  = params.GetVendorId();
+        if (productId.has_value() && vendorId.has_value())
+        {
+            snprintf(txtVidPid, sizeof(txtVidPid), "VP=%d+%d", *vendorId, *productId);
+            txtFields[numTxtFields++] = txtVidPid;
+        }
+        else if (vendorId.has_value())
+        {
+            snprintf(txtVidPid, sizeof(txtVidPid), "VP=%d", *vendorId);
+            txtFields[numTxtFields++] = txtVidPid;
+        }
     }
 
     char txtDeviceType[chip::Dnssd::kKeyDeviceTypeMaxLength + 4];
-    if (params.GetDeviceType().HasValue())
     {
-        snprintf(txtDeviceType, sizeof(txtDeviceType), "DT=%" PRIu32, params.GetDeviceType().Value());
-        txtFields[numTxtFields++] = txtDeviceType;
+        const auto & deviceType = params.GetDeviceType();
+        if (deviceType.has_value())
+        {
+            snprintf(txtDeviceType, sizeof(txtDeviceType), "DT=%" PRIu32, *deviceType);
+            txtFields[numTxtFields++] = txtDeviceType;
+        }
     }
 
     char txtDeviceName[chip::Dnssd::kKeyDeviceNameMaxLength + 4];
-    if (params.GetDeviceName().HasValue())
     {
-        snprintf(txtDeviceName, sizeof(txtDeviceName), "DN=%s", params.GetDeviceName().Value());
-        txtFields[numTxtFields++] = txtDeviceName;
+        const auto deviceName = params.GetDeviceName();
```

Comment:
```suggestion
        const auto& deviceName = params.GetDeviceName();
```
to make it consistent with other fields.